lib/responder.js

```diff
@@ -1156,6 +1157,17 @@ function operationResultStream() {
 function operationErrorListener(error) {
   /*jshint validthis:true */
   var operation = this;
+  if(operation.client.connectionParams.apiKey){
+    if(error.statusCode === 403 && operation.expiration <= (new Date())){
+      if(!operation.lockAccessToken){
```

Comment:
I am now wondering if anything is needed - according to https://stackoverflow.com/a/7238663/3306099 , JavaScript doesn't have the same issues as e.g. Java where multiple threads can access the same block of code at the same time. If that's the case, then `lockAccessToken` isn't needed - the code can just say "If I get a 403, then I'll generate a new token and try again".

Reply:
That's correct. This condition will be helpful for DMSDK where we are creating multiple threads.